x/ibc/23-commitment/types/merkle.go

```diff
@@ -336,8 +336,8 @@ func verifyChainedMembershipProof(root []byte, specs []*ics23.ProofSpec, proofs
 		// from the proofs and specs which are lowest to highest
 		key := keys.GetKey(-1 * (i + 1))
 		if ok := ics23.VerifyMembership(specs[i], subroot, proofs[i], key, value); !ok {
-			return sdkerrors.Wrapf(ErrInvalidProof, "chained membership proof failed to verify membership of value: %X in subroot %X at index %d",
-				value, subroot, i)
+			return sdkerrors.Wrapf(ErrInvalidProof, "chained membership proof failed to verify membership of value: %X in subroot %X at index %d for proof %v",
+				value, subroot, i, proofs[i])
```

Comment:
I'm not sure how to return error for this, but various parts of the proof are useful in the error, like what type of proof this is (existence, non-existence) and what the key is. 

Given that the purpose of this package is verification of many different pieces, I think providing some higher level context on what actually failed will be useful for people not familiar with proof structures